aesthetic/avg_autotags.py

```diff
@@ -23,51 +23,80 @@
+        #put in 'NUM_PHOTOS', autotags_90plus to overall_autotags_90plus
         overall_autotags_90plus.update(autotags_90plus + ['NUM_PHOTOS'])
- 
-    
+        # this looks like Counter({'NUM_PHOTOS': 15, 'outdoor': 5, 'building':
+        # 2, 'gable': 2, 'nature': 2, 'indoor': 2, 'plant': 2, 'architecture':
+        # 2, 'pink': 1, 'vehicle': 1, 'house': 1, 'pediment': 1, 'frieze': 1,
+        # 'flower': 1, 'green': 1, 'suit': 1, 'banana tree': 1, 'car': 1})
+
     # Compute the overall rates of each tag, so we can subtract them off.
     tempdict = collections.Counter()
     for tag, value in overall_autotags_90plus.items():
         tempdict[tag] = (value*1.0/overall_autotags_90plus['NUM_PHOTOS'])
     del tempdict['NUM_PHOTOS']
     overall_autotags_90plus = tempdict
-
-    output = {}
+    sorted_output = dict()
+    output = dict()
     for nghd, autotags90plus in nghds_autotags90plus.items():
+    	sorted_output[nghd] = dict() #this is output with the max 10 autotags
+        sorted_output[nghd]["autotags_90plus_minusbaseline"] = list()
         autotags = nghds_autotags[nghd]
+        # Counter({'NUM_PHOTOS': 1, 'indoor': 0.925, 'people': 0.895, 'blue':
+        # 0.695, 'sleeping': 0.674})
         num_photos = autotags90plus['NUM_PHOTOS']
-        output[nghd] = {'num_photos': num_photos, 'autotags_90plus': {},
-                'autotags_all': {}, 'autotags_90plus_minusbaseline':{}}
+        sorted_output[nghd]['NUM_PHOTOS'] = num_photos
+        output[nghd] = {'num_photos': num_photos, #'autotags_90plus': {}, 'autotags_all': {}, 
+        'autotags_90plus_minusbaseline':{}}
             # 'autotags_90plus': [], 'autotags_all': []}
         for autotag, count in autotags90plus.most_common():
             if autotag == 'NUM_PHOTOS':
                 continue
             pct_photos = round(count * 1.0 / num_photos, 5)
-            output[nghd]['autotags_90plus'][autotag] = pct_photos
+            # output[nghd]['autotags_90plus'][autotag] = pct_photos
             # output[nghd]['autotags_90plus'].append((autotag, round(count*1.0/num_photos, 5)))
+            # for each neighborhood, find the max 10 autotags
             output[nghd]['autotags_90plus_minusbaseline'][autotag] = round(pct_photos - overall_autotags_90plus[autotag], 5)
-
-        for autotag, count in autotags.most_common():
-            if autotag == 'NUM_PHOTOS':
+        #figiure out the top 10 autotags for each output[neighborhood]['autotags_90plus_minusbaseline'] which is a dict of tag:pct
+        sorted_output[nghd]['autotags_90plus_minusbaseline'] = dict(sorted(output[nghd]['autotags_90plus_minusbaseline'].iteritems(), key=operator.itemgetter(1), reverse=True)[:10])
+   
+    ###-------now add the urls to sorted output -------------
+    # this is the final output, with sorted top 10 autotags and urls
+    output_autotag_url = dict()
+    for nghd in sorted_output: 
+        output_autotag_url[nghd] = dict()
```

Comment:
You can just declare all this in one line: 

    output_autotag_url[nghd] = {'autotags_90plus_minusbaseline': [], 'NUM_PHOTOS': sorted_output[nghd]['NUM_PHOTOS']}